src/riak_kv_bucket.erl

```diff
@@ -498,103 +638,76 @@ prop_create_valid() ->
-                               %% typed bucket, strongly-consistent, both n_val and consistent value are invalid changes
-                               {false, _, Mult,_,true,true,false} ->
-                                   maybe_bad_mult(Mult,
-                                                  merge(lists:keydelete(consistent, 1, lists:keydelete(n_val, 1, New)), Existing));
-                               %% typed bucket, strongly-consistent, both n_val and consistent value are invalid changes
-                               {false, _, Mult,_,true,true,notvalid} ->
-                                   maybe_bad_mult(Mult,
-                                                  merge(lists:keydelete(consistent, 1, lists:keydelete(n_val, 1, New)), Existing));
-                               %% typed bucket, strongly-consistent, n_val change is invalid
-                               {false, _, Mult,_,true,true,_} ->
-                                   maybe_bad_mult(Mult, merge(lists:keydelete(n_val, 1, New), Existing));
-                               %% typed bucket, strongly-consistent, consistent value change is invalid
-                               {false, _, Mult, _, _, true, false} ->
-                                   maybe_bad_mult(Mult, merge(lists:keydelete(consistent, 1, New), Existing));
-                               %% typed bucket, strongly-consistent, consistent value change is invalid
-                               {false, _, Mult, _, _, true, notvalid} ->
-                                   maybe_bad_mult(Mult, merge(lists:keydelete(consistent, 1, New), Existing));
-                               %% typed bucket, strongly-consistent, all good (except maybe allow_mult)
-                               {false, _, Mult, _, _, true, _} ->
-                                   maybe_bad_mult(Mult, merge(New, Existing));
-                               %% typed bucket, strongly-consistent, all good (except maybe allow_mult)
-                               {false, true, _Mult, true,_, _, _} ->
-                                   NVal = proplists:get_value(n_val, New, proplists:get_value(n_val, Existing)),
-                                   merge([{n_val, NVal}], Existing);
-                               %% typed bucket, bucket not strongly consistent or a data type, all valid
-                               {false,_,_,_,_,false,_} ->
-                                   merge(New, Existing)
-                           end,
-
+                Result={Good, Bad} = validate(update, Bucket, Existing, New),
+
+                %% All we really want to check is that every key in
+                %% Good replaces the same key in Existing, right?
+                %% Remove `Bad' from the inputs to validate.
+                {NoBadExisting, OnlyGoodNew} = lists:foldl(fun({Name, _Err}, {Old, Neu}) ->
+                                                                   {value, V, Neu2} = lists:keytake(Name, 1, Neu),
```

Comment:
The line wrapping here is ridiculous. Could you put the anonymous function above the fold call, or make it a module function? It doesn't look like it needs any bindings from the surrounding code.